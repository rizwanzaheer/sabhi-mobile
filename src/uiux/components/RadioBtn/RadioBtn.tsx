import React from 'react'
import { TouchableHighlight } from 'react-native'
import Container from '../Container/Container'
import Icon from '../Icon/Icon'
import Text from '../Text/Text'
import { withTheme } from '../../theming'

interface RadioProps {
  /**
   *  Pass a test ID to button
   */
  testID?: string
  /**
   * On Press handler
   */
  onPress: () => void

  /**
   * Selected flag
   */
  selected: boolean

  /**
   * Disable the button
   */
  disabled?: boolean

  /**
   * Theme
   */
  theme: any
}

const Radio: React.FC<RadioProps> = props => {
  return (
    <TouchableHighlight
      onPress={props.onPress}
      underlayColor={'transparent'}
      disabled={props.disabled}
      testID={props.testID}
    >
      <Container flexDirection={'row'} alignItems={'center'} marginBottom={5}>
        <Icon icon={props.selected ? props.theme.icons.RADIO_ON : props.theme.icons.RADIO_OFF} />
        <Container flex={1} paddingLeft={8}>
          <Text>{props.children}</Text>
        </Container>
      </Container>
    </TouchableHighlight>
  )
}

export default withTheme(Radio)
