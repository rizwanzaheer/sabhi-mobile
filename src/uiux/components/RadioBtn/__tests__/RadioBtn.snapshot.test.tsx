import * as React from 'react'
import { render } from 'react-native-testing-library'
import RadioBtn from '../RadioBtn'

jest.mock('react-native-vector-icons/Ionicons', () => 'Icon')

describe('Component(snapshot): Indicator', () => {
  it('renders with default props', () => {
    const onPress = jest.fn()
    const tree = render(<RadioBtn selected={true} onPress={onPress} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders unselected', () => {
    const onPress = jest.fn()
    const tree = render(<RadioBtn selected={false} onPress={onPress} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders disabled', () => {
    const onPress = jest.fn()
    const tree = render(<RadioBtn selected={false} onPress={onPress} disabled={true} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with children text', () => {
    const onPress = jest.fn()
    const tree = render(
      <RadioBtn selected={false} onPress={onPress}>
        {'Test radio button text'}
      </RadioBtn>,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
