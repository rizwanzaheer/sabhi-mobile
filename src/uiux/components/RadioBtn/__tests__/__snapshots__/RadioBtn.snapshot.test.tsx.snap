// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(snapshot): Indicator renders disabled 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": "row",
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": 5,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <Icon
      color="#BBBBBB"
      name="ios-radio-button-off"
      size={26}
    />
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": 1,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": 8,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "flexWrap": "wrap",
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Component(snapshot): Indicator renders unselected 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": "row",
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": 5,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <Icon
      color="#BBBBBB"
      name="ios-radio-button-off"
      size={26}
    />
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": 1,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": 8,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "flexWrap": "wrap",
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Component(snapshot): Indicator renders with children text 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": "row",
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": 5,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <Icon
      color="#BBBBBB"
      name="ios-radio-button-off"
      size={26}
    />
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": 1,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": 8,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "flexWrap": "wrap",
          }
        }
      >
        Test radio button text
      </Text>
    </View>
  </View>
</View>
`;

exports[`Component(snapshot): Indicator renders with default props 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": "row",
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": 5,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <Icon
      color="#BBBBBB"
      name="ios-radio-button-on"
      size={26}
    />
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": 1,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": 8,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "flexWrap": "wrap",
          }
        }
      />
    </View>
  </View>
</View>
`;
